sdk/lib/rtl/registry.c

```diff
@@ -330,8 +330,8 @@ RtlpCallQueryRegistryRoutine(IN PRTL_QUERY_REGISTRY_TABLE QueryTable,
                 RtlInitEmptyUnicodeString(&Source, Data, (USHORT)Length);
                 Source.Length = Source.MaximumLength - sizeof(UNICODE_NULL);
 
-                /* Setup the desination string */
-                RtlInitEmptyUnicodeString(&Destination, (PWCHAR)SpareData, 0);
+                /* Setup the destination string */
+                RtlInitEmptyUnicodeString(&Destination, ((PWCHAR)Data + Length + sizeof(UNICODE_NULL)), 0);
```

Comment:
The unit of `PWCHAR` is `WCHAR`.
Adding `sizeof(UNICODE_NULL) * sizeof(WCHAR)` bytes is questionable.

Reply:
Also `Length` is in bytes. And you can use `Source.MaximumLength`, which already contains the `UNICODE_NULL`:
`RtlInitEmptyUnicodeString(&Destination, (PWCHAR)((PCHAR)Data + Source.MaximumLength), 0);`
or
`RtlInitEmptyUnicodeString(&Destination, Add2Ptr(Data, Length + Source.MaximumLength), 0);`